client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -345,4 +358,66 @@ public void close() {
             this.connectionPool.close();
         }
     }
+
+    @Override
+    public CompletableFuture<Long> getDistanceBetweenTwoStreamCuts(Stream stream, StreamCut fromStreamCut, StreamCut toStreamCut) {
+        Preconditions.checkNotNull(fromStreamCut, "fromStreamCut");
+        Preconditions.checkNotNull(toStreamCut, "toStreamCut");
+        if (!fromStreamCut.equals(StreamCut.UNBOUNDED) && !toStreamCut.equals(StreamCut.UNBOUNDED)) {
+            Preconditions.checkArgument(fromStreamCut.asImpl().getStream().equals(toStreamCut.asImpl().getStream()),
```

Comment:
This checks to be sure they are the same streamCut as each other but not if each is the same as `stream`.